syft/pkg/cataloger/binary/default_classifiers.go

```diff
@@ -29,11 +29,39 @@ var defaultClassifiers = []classifier{
 	{
 		Class:    "go-binary",
 		FileGlob: "**/go",
-		EvidenceMatcher: fileContentsVersionMatcher(patternEndingWithNull(
-			`(?m)go(?P<version>[0-9]+\.[0-9]+(\.[0-9]+|beta[0-9]+|alpha[0-9]+|rc[0-9]+)?)`)),
+		EvidenceMatcher: fileContentsVersionMatcher(
+			`(?m)go(?P<version>[0-9]+\.[0-9]+(\.[0-9]+|beta[0-9]+|alpha[0-9]+|rc[0-9]+)?)\x00`),
 		Package: "go",
 		CPEs:    singleCPE("cpe:2.3:a:golang:go:*:*:*:*:*:*:*:*"),
 	},
+	{
+		Class:    "java-binary-openjdk",
+		FileGlob: "**/java",
+		EvidenceMatcher: fileContentsVersionMatcher(
+			// [NUL]openjdk[NUL]java[NUL]1.8[NUL]1.8.0_352-b08[NUL]
+			`(?m)\x00openjdk\x00java\x00(?P<release>[0-9]+[.0-9]+)\x00(?P<version>[0-9]+[-._a-zA-Z0-9]+)\x00`),
+		Package: "java",
+		// TODO the updates might need to be part of the CPE, like: 1.8.0:update152
```

Comment:
Does this TODO still apply? Are there any changes we have to make in the CPE generation logic?

Reply:
@spiffs yes I think the TODO is still possibly applicable, but for now I think it's okay to just surface the entire version string because that's what the package managers are doing. I think they will _all_ have the same potential problem matching against the CPEs and I'd rather find a solution for all java packages instead of just these classifiers. It needs more investigation